generator/generator.py

```diff
@@ -26,15 +26,16 @@ def add_ports(self, **kwargs):
     def wire(self, port0, port1):
         assert isinstance(port0, PortReferenceBase)
         assert isinstance(port1, PortReferenceBase)
-        self.wires.append((port0, port1))
+        connection = (max(port0, port1), min(port0, port1))
+        if connection not in self.wires:
+            self.wires.append(connection)
```

Comment:
can we print a warning here in the `else`: "skipping duplicate connection: {port0.name}, {port1.name}"

Reply:
Done